### PR TITLE
Add ability to add additional hibernate dialect for third party databases. 

### PR DESCRIPTION
**Pull request by alexeysharandin**
At this moment all Hibernates dialects hardcoded in io.quarkus.hibernate.orm.deployment.Dialects

This PR first step of resolving comment in this class:
        // For now select the latest dialect from the driver
        // later, we can keep doing that but also avoid DCE
        // of all the dialects we want in so that people can override them

### CODE REVIEW

**Review by yrodiere — APPROVED** [review]
Thanks!

I added a few commits, mainly to:

1. Move the newly introduced build item to a dedicated module for SPIs of the Hibernate ORM extension, because this really is about Hibernate ORM and not datasources/jdbc.
2. Use the newly introduced build item for all core DB kinds. The main advantage is that we get testing of that build item for free, thanks to existing integration tests.

Unless @Sanne or @gsmet objects to the creation of a new module, let's merge this once CI agrees.

**Comment by Sanne** [comment]
sounds good - no objections! Thanks :)

**Comment by yrodiere** [comment]
Merged! Thanks again @alexeysharandin , and good luck with your extension :)

**Comment by gastaldi** [comment]
That is really cool! But the build item name isn't too friendly (`DatasourceDbKindHibernateOrmMetadataBuildItem` 😅  ) for two reasons:
- The class is already in a package with Hibernate in its name, I don't see why adding Hibernate in the name would help;
- This is just a DbKind/Dialect tuple. If another metadata specific build item is necessary, the constructor may need to be changed or a builder would need to be introduced.
 
How about naming it simply `io.quarkus.hibernate.orm.deployment.spi.DatabaseKindDialectBuildItem`?

File: extensions/hibernate-orm/deployment-spi/pom.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-core-deployment</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-datasource-common</artifactId>
+        </dependency>
```
Comment (gastaldi) [inline extensions/hibernate-orm/deployment-spi/pom.xml:23]: This can be removed as it's not used at the moment

**Comment by yrodiere** [comment]
> UPDATE: Just saw that the original class was io.quarkus.datasource.deployment.spi.DatabaseKindBuildItem. Better add an entry in the migration guide about that.

That is a different build item. We didn't rename an existing build item, we added another one specifically for Hibernate ORM metadata.

> This is just a DbKind/Dialect tuple. If another Hibernate ORM metadata specific attribute is necessary, the constructor may need to be changed or a builder would need to be introduced.

Yes we would need another constructor. Which is much easier to do in a backwards compatible way than renaming the class. And that's something we've done multiple times in the Hibernate ORM extension.

> The class is already in a package with Hibernate in its name, I don't see why adding Hibernate in the name would help;

That I can agree with : the name is very long 😀 Though I'm not sure the renaming is worth the trouble now that it's been merged and @alexeysharandin is probably already trying to use it.

**Comment by alexeysharandin** [comment]
@yrodiere don't worry about me and naming. I'm waiting when this PR will be included to Quarkus 2.x and only after this can use this BuildItem. In this case - naming up to you :)

**Pull request merged**